File: chart/src/main/java/cn/jingzhuan/lib/chart/data/BarValue.java
package cn.jingzhuan.lib.chart.data;

import android.graphics.Paint;

/**
 * Created by Donglua on 17/8/1.
 */
public class BarValue extends Value {

    private float[] yValues;

    private int mColor = -2;

    private Paint.Style mPaintStyle = Paint.Style.FILL;

    private boolean isEnable = true;
    private int[] gradientColors = null;

    public BarValue(float[] yValues) {
        this.yValues = yValues;
    }

    public BarValue(float[] yValues, int color) {
        this.yValues = yValues;
        this.mColor = color;
    }

    public BarValue(float value1, float value2, int color) {
        this.yValues = new float[] {value1, value2};
        this.mColor = color;
    }

    public BarValue(float value1, float value2, int color, Paint.Style paintStyle) {
        this.yValues = new float[] {value1, value2};
        this.mColor = color;
        this.mPaintStyle = paintStyle;
    }

    public BarValue(float value, int color, Paint.Style paintStyle) {
        this.yValues = new float[] {value, 0f};
        this.mColor = color;
        this.mPaintStyle = paintStyle;
    }

    public BarValue(float[] values, int color, Paint.Style paintStyle) {
        this.yValues = values;
        this.mColor = color;
        this.mPaintStyle = paintStyle;
    }

    public BarValue(float yValue) {
        this.yValues = new float[] { yValue, 0f };
    }

    public BarValue(float yValue, int color) {
        this.yValues = new float[] { yValue, 0f };
        this.mColor = color;
    }

    public float[] getValues() {
        return yValues;
    }

    public void setValues(float[] yValues) {
        this.yValues = yValues;
    }

    public void setValues(float yValue) {
        this.yValues = new float[] { yValue, 0f };
    }

    public int getValueCount() {

        if (yValues == null) return 0;

        return yValues.length;
    }

    public int getColor() {
        return mColor;
    }

    public void setColor(int color) {
        mColor = color;
    }

    public void setPaintStyle(Paint.Style mPaintStyle) {
        this.mPaintStyle = mPaintStyle;
    }

    public Paint.Style getPaintStyle() {
        return mPaintStyle;
    }

    public void setEnable(boolean enable) {
        isEnable = enable;
    }

    public boolean isEnable() {
        return isEnable;
    }

    public void setGradientColors(int colorTop, int colorBottom) {
        this.gradientColors = new int[] {colorTop, colorBottom};
    }

    public int[] getGradientColors() {
        return gradientColors;
    }
}

File: chart/src/main/java/cn/jingzhuan/lib/chart3/data/value/BarValue.kt
package cn.jingzhuan.lib.chart3.data.value

import android.graphics.Paint

/**
 * @since 2023-09-05
 * created by lei
 */
class BarValue : AbstractValue {
    var values: DoubleArray? = null

    var color = -2

    var paintStyle = Paint.Style.FILL

    var isEnable = true

    var gradientColors: IntArray? = null

    /**
     * 这里tag比dataSet的优先级高
     */
    var tag: String? = null

    val valueCount: Int
        get() = values?.size ?: 0

    // 兼容 double
    constructor(yValues: DoubleArray?) {
        values = yValues
    }

    constructor(yValues: DoubleArray?, color: Int) {
        values = yValues
        this.color = color
    }

    constructor(value1: Double, value2: Double, color: Int) {
        values = doubleArrayOf(value1, value2)
        this.color = color
    }

    constructor(value1: Double, value2: Double, color: Int, paintStyle: Paint.Style) {
        values = doubleArrayOf(value1, value2)
        this.color = color
        this.paintStyle = paintStyle
    }

    constructor(value: Double, color: Int, paintStyle: Paint.Style) {
        values = doubleArrayOf(value, 0.0)
        this.color = color
        this.paintStyle = paintStyle
    }

    constructor(values: DoubleArray?, color: Int, paintStyle: Paint.Style) {
        this.values = values
        this.color = color
        this.paintStyle = paintStyle
    }

    constructor(yValue: Double) {
        values = doubleArrayOf(yValue, 0.0)
    }

    constructor(yValue: Double, color: Int) {
        values = doubleArrayOf(yValue, 0.0)
        this.color = color
    }

    fun setValues(yValue: Double) {
        values = doubleArrayOf(yValue, 0.0)
    }

    // ------ 兼容 Float 类型重载 ------
    constructor(yValues: FloatArray?) : this(yValues?.map { it.toDouble() }?.toDoubleArray())
    constructor(yValues: FloatArray?, color: Int) : this(yValues?.map { it.toDouble() }?.toDoubleArray(), color)
    constructor(value1: Float, value2: Float, color: Int) : this(value1.toDouble(), value2.toDouble(), color)
    constructor(value1: Float, value2: Float, color: Int, paintStyle: Paint.Style) : this(value1.toDouble(), value2.toDouble(), color, paintStyle)
    constructor(value: Float, color: Int, paintStyle: Paint.Style) : this(value.toDouble(), color, paintStyle)
    constructor(values: FloatArray?, color: Int, paintStyle: Paint.Style) : this(values?.map { it.toDouble() }?.toDoubleArray(), color, paintStyle)
    constructor(yValue: Float) : this(yValue.toDouble())
    constructor(yValue: Float, color: Int) : this(yValue.toDouble(), color)

    fun setValues(yValue: Float) = setValues(yValue.toDouble())
    // -------------------------------

    fun setGradientColors(colorTop: Int, colorBottom: Int) {
        gradientColors = intArrayOf(colorTop, colorBottom)
    }
}
